### DISCUSSION TITLE
Can a superuser create a new user and/or get all users?

### DISCUSSION THREAD

**Discussion by felipebustillo** [fastapi-users/fastapi-users#972] [Q&A] [answered]
Maybe I missed this one, but I was looking for some route that, if you're a superuser, lets you create a new user from scratch.

I tried this, but I get errors all the time and it conflicts with pydantic:

```
# Create user
@users_router.post("/users", tags=["users"], dependencies=[Depends(allowed_roles)])
async def create_user(user: UserCreate = Body(...)):
    user = jsonable_encoder(user)
    new_user = await db["users"].insert_one(user)
    created_user = await db["users"].find_one({"_id": new_user.inserted_id})
    return JSONResponse(status_code=status.HTTP_201_CREATED, content=created_user)
```

Regarding the read operation for users, I did this:

```
# Get users
@users_router.get("/users", tags=["users"], response_model=Page[UserDB], dependencies=[Depends(allowed_roles)])
async def get_users():
    users = await db["users"].find().to_list(1000)
    return paginate(users)


add_pagination(users_router)
```

**Comment by alwaz-shahid** [comment]
Currently, I'm too developing a multi-tenant API using fastapi and utilizing fastapi-users,  for the user creation  for now I'm considering restricting access to the register route, thus I'll have to inject a root user in the database  manually. I'm still looking for a more reliable solution.

**Comment by frankie567** [comment, marked as answer]
If you operate inside FastAPI context, the easiest way is to use the `user_manager.create` method. It's here for that. Here is how we are doing it in the `register` route, you can take inspiration from there:

https://github.com/fastapi-users/fastapi-users/blob/2477fdffc118f9d75b9031f236037e57e058dbaa/fastapi_users/router/register.py#L51-L74

---

If you need to do it programmatically outside FastAPI, e.g. in a CLI, there is a recipe for that in the docs: https://fastapi-users.github.io/fastapi-users/10.0/cookbook/create-user-programmatically/
